CRB-238 Dockerfile and image build updates

will-gluwa opened the pull request:
- Use dotnet SDK image as base (from https://hub.docker.com/_/microsoft-dotnet-sdk)
- Add build.sh script to build docker image from WSL/Linux

Commit:
Dockerfile and image build updates
- Use dotnet SDK image as base (from https://hub.docker.com/_/microsoft-dotnet-sdk)
- Add build.sh script to build docker image from WSL/Linux

atodorov commented on build.sh:
```diff
@@ -0,0 +1,6 @@
+rm -rf ccgateway/bin/Release/netcoreapp3.1/plugins
+mkdir ccgateway/bin/Release/netcoreapp3.1/plugins
+cp ccgateway/plugins/*.dll ccgateway/bin/Release/netcoreapp3.1/plugins/
+docker build --tag creditcoin-gateway ccgateway/bin/Release/netcoreapp3.1/.
```
`.github/workflows/dotnet.yml` should at least call this script to make sure we don't result in errors.

will-gluwa replied:
Implement build purely via dockerfile and updated github actions to test docker build

atodorov commented on build.sh:
```diff
@@ -0,0 +1,6 @@
+dotnet build --configuration Release
+rm -rf ccgateway/bin/Release/netcoreapp3.1/plugins
+mkdir ccgateway/bin/Release/netcoreapp3.1/plugins
+cp ccgateway/plugins/*.dll ccgateway/bin/Release/netcoreapp3.1/plugins/
```
I'd prefer these steps to be inside of the Dockerfile so that no matter on what system we execute this script we get the same results and everything is self-contained inside the container? What do you think about that ?

Note: maybe this is opening up a bigger discussion than originally intended by the Jira ticket.

colingluwa replied:
Looks good,
Just wondering if we should migrate the output to another directory. We should be copying the framework out to a running destination like. /opt/ccgateway.

will-gluwa replied:
@atodorov Moved the steps directly to the dockerfile

atodorov reviewed (commented):
Added some comments.

Commit:
Use dockerfile only; update CI
- Updated implementation to just use Dockerfile (no build.sh)
- Updated github actions
